### PR TITLE
Try to detect when certs store is out of sync with control plane

### PR DESCRIPTION
**Pull request by primeroz**
Not even tested yet

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -108,6 +152,16 @@ func block(host string) {
...
 		}
 
+		// If Enabled perform the ENVOY ISTIO SDS WARMING Check
+		sdsprotection, ok := os.LookupEnv("ENVOY_ISTIO_SDS_WARMING_PROTECTION")
+		if ok && sdsprotection == "true" {
+			err := checkEnvoyIstioSDS(host)
+			if err != nil {
+				return err
+			}
+			return errors.New("Envoy Istio SDS check failed")
```
Comment (GeorgeMac) [inline main.go]: Looks like you error in all cases here. Is that what you want?

File: main.go [hunk 2]
```diff
@@ -83,6 +102,31 @@ func main() {
...
+
+	certs := &Certificates{}
+
+	err := rsp.Decode(certs)
+	if err != nil {
+		return err
+	}
+
+	// Check number of CERTS is > 1, if not hit the KILL Endpoint
+	if len(certs.Certificates) == 1 {
```
Comment (GeorgeMac) [inline main.go]: Perhaps `len(certs.Certificates) < 2` ? to catch the 0 case? I assume that is also an error?

File: main.go (hunk 1 above)
Comment (primeroz) [reply main.go]: thanks, this should be fixed now

File: main.go (hunk 2 above)
Comment (primeroz) [reply main.go]: done

**Pull request closed without merging**